src/containerapp/azext_containerapp/_clients.py

```diff
@@ -70,6 +72,53 @@ def poll(cmd, request_url, poll_if_status):  # pylint: disable=inconsistent-retu
             raise e
 
 
+def poll_status(cmd, request_url):  # pylint: disable=inconsistent-return-statements
+    from azure.core.exceptions import HttpResponseError
+    start = time.time()
+    end = time.time() + POLLING_TIMEOUT
+    animation = PollingAnimation()
+
+    animation.tick()
+    r = send_raw_request(cmd.cli_ctx, "GET", request_url)
+
+    while r.status_code in [200] and start < end:
+        time.sleep(POLLING_SECONDS)
+        animation.tick()
+        r = send_raw_request(cmd.cli_ctx, "GET", request_url)
+        response_body = json.loads(r.text)
+        if response_body["status"].lower() in ["failed", "canceled"]:
+            message = json.dumps(response_body["error"]) if "error" in response_body else "Operation failed or canceled"
+            raise HttpResponseError(
+                response=r,
+                message=message
+            )
+        if response_body["status"].lower() in ["succeeded"]:
+            break
+        start = time.time()
+
+    animation.flush()
+    return
+
+
+def poll_results(cmd, request_url):  # pylint: disable=inconsistent-return-statements
+    start = time.time()
+    end = time.time() + POLLING_TIMEOUT
+    animation = PollingAnimation()
+
+    animation.tick()
+    r = send_raw_request(cmd.cli_ctx, "GET", request_url)
+
+    while r.status_code in [202] and start < end:
```

Comment:
what about the status code is error status code like 400?

Reply:
if status code not OK , send_raw_request will throw exception.